src/Storages/ExternalStream/Kafka/Kafka.cpp

```diff
@@ -64,10 +217,25 @@ Kafka::Kafka(IStorage * storage, std::unique_ptr<ExternalStreamSettings> setting
         settings->set("one_message_per_row", true);
     }
 
+    if (!context->getSettingsRef().shards.value.empty())
+        shards_from_settings = parseShards(context->getSettingsRef().shards.value);
```

Comment:
we should support create a topic with specified shards num (i.e. partition num)

Reply:
No, the `rd_kafka_topic_t` object is supposed to be shared with all paritions. Each KafkaSource instance use the same topic to call `startConsume` with their own parition number.